intact-orthology-import/src/main/java/uk/ac/ebi/intact/ortholog/OrthologsProteinAssociation.java

```diff
@@ -152,4 +152,22 @@ public void dataWriter(String toWrite) {
             e.printStackTrace(); // Print the stack trace to help with debugging
         }
     }
+
+    public static Collection<String> associateOneProteinToPantherIds(String filePath, IntactProtein protein) throws IOException {
```

Comment:
This method get all the panther ids for a given protein from the file with the pairs.
It is slower than relying on the map, but we won't run into memory issues.

Reply:
A way to make this more efficient.

1. Change `OrthologsFileParser.parseFileAndSave` to instead of taking an `outputFilePath` parameters, it takes and `outputDirPath` parameter. Then, instead of saving all the pairs in a single file, we save all the pairs in the same directory, with an individual file per protein.

2. Change this method to read directly from the file for the specific protein. Then we only need to check if the file exists, and get the panther ids from it.

Reply:
In order to make it more efficient:

1. Add this dependency
   ```
          <dependency>
            <groupId>commons-io</groupId>
            <artifactId>commons-io</artifactId>
            <version>2.4</version>
        </dependency>
   ```

2. Update `OrthologsFileParser` methods, `parseFileAndSave` and `writePair` to deal with a directory and multiple files
   ```
    public static void parseFileAndSave(String inputFilePath, String outputDirPath) throws IOException {
        Pattern uniprotKBRegex = Pattern.compile("UniProtKB=([A-Z0-9]+)");
        Pattern pantherRegex = Pattern.compile("PTHR\\d+");
        long uniprotAndPantherCount = 0;
        log.info("Parsing file...");

        File outputDir = new File(outputDirPath);
        // First, we empty de directory to start clean
        if (outputDir.exists()) {
            FileUtils.deleteDirectory(outputDir);
        }
        outputDir.mkdirs();

        try (BufferedReader reader = new BufferedReader(new FileReader(inputFilePath))) {
            ...
                while (pantherMatcher.find()) {
                    for (String uniprotMatch : uniprotMatches) {
                        writePair(outputDir.toPath(), uniprotMatch, pantherMatcher.group());
                        uniprotAndPantherCount += uniprotMatches.size();
                    }
                }
           ...
    private static void writePair(Path dirPath, String uniprotId, String pantherId) throws IOException {
        Path filePath = dirPath.resolve(uniprotId);
        try (FileWriter fileWriter = new FileWriter(filePath.toFile(), true);
            ...
    ```

3. Update `OrthologsProteinAssociation` method `associateOneProteinToPantherIds` to read from the directory
   ```
    public static Collection<String> associateOneProteinToPantherIds(String dirPath, IntactProtein protein) throws IOException {
        Path filePath = Path.of(dirPath).resolve(protein.getUniprotkb());
        List<String> pantherIds = new ArrayList<>();
        try (BufferedReader reader = new BufferedReader(new FileReader(filePath.toFile()))) {
           ...
   ```

4. Update `orthology-import.properties`
   ```
      #proteinPantherPairFilePath=ProteinAndOrthologPairs.txt
      proteinPantherPairDirPath=ProteinAndOrthologPairs
   ```

5. Replace `proteinPantherPairFilePath` with `proteinPantherPairDirPath` everywhere.